satellite/marketing/marketingweb/server.go

```diff
@@ -70,31 +102,74 @@ func NewServer(logger *zap.Logger, config Config, listener net.Listener) *Server
 	return s
 }
 
-// ServeHTTP handles index request
-func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+// Serves index page and renders offer and credits tables
+func (s *Server) getOffers(w http.ResponseWriter, req *http.Request){
 	if req.URL.Path != "/" {
 		s.serveNotFound(w, req)
 		return
 	}
 
+	offers, err := s.service.ListAllOffers(context.Background())
+	if err != nil {
+		s.log.Error("app handler error", zap.Error(err))
+		s.serveInternalError(w,req,err)
+		return
+	}
+
 	files := append(s.commonPages(),
 		filepath.Join(s.templateDir, "home.html"),
 		filepath.Join(s.templateDir, "refOffers.html"),
 		filepath.Join(s.templateDir, "freeOffers.html"),
 		filepath.Join(s.templateDir, "roModal.html"),
 		filepath.Join(s.templateDir, "foModal.html"),
 	)
-
-	home, err := template.New("landingPage").ParseFiles(files...)
+	home := template.Must(template.New("landingPage").ParseFiles(files...))
+	err = home.ExecuteTemplate(w, "base", organizeOffers(offers))
 	if err != nil {
-		s.serveInternalError(w, req)
+		s.serveInternalError(w,req,err)
 		return
 	}
+}
 
-	err = home.ExecuteTemplate(w, "base", nil)
+// Helper function used by createOffer to turn post form input into a new offer
+func (s *Server) formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error){
+	err := req.ParseForm()
 	if err != nil {
-		s.log.Error("failed to execute template", zap.Error(err))
+		return o, err
+	}
+	defer req.Body.Close()
+
+	decoder.RegisterConverter(time.Time{}, timeConverter)
```

Comment:
This might race.

Reply:
Is it because decoder is global ?

Reply:
https://github.com/storj/storj/pull/2186#discussion_r293196093

Reply:
The globalness wasn't the problem, it was that it was calling `Register` from a func that is called concurrently. If it were done in an init func it might have been fine.

Reply:
I see. I will move the `decoder` var back in the global vars block since I need it in an init func for registering the `timeConverter` and also `formToStruct ()` for the actual decoding.

I will use the decoder to register the timeConverter in the `NewServer` whee it should be safe no?

Reply:
If this is good I pushed it already.

Reply:
No, in an init func... so:

```
func init() {
   decoder.Register(...
```

Reply:
Gotcha. I understood this better after reading [this](https://tutorialedge.net/golang/the-go-init-function/.) I pushed the change